pkg/trainer/training.go

```diff
@@ -275,77 +242,62 @@ func (j *TrainingJob) masterName() string {
 }
 
 // setup the training job.
-func (j *TrainingJob) setup(config *spec.ControllerConfig) error {
+func (j *TrainingJob) setup(config *spec.ControllerConfig) {
 	if j.job == nil {
-		return fmt.Errorf("job.Spec can't be nil")
-	}
-
-	err := j.job.Spec.SetDefaults()
-	if err != nil {
-		return fmt.Errorf("there was a problem setting defaults for job spec: %v", err)
+		j.status.SetReason("Internal error; tried to setup a job with no spec.")
```

Comment:
maybe `tried to` --> `failed to` would be better

Reply:
Done.